src/lib/sw_engine/tvgSwRle.cpp

```diff
@@ -613,6 +613,94 @@ static int _genRle(RleWorker& rw)
 }
 
 
+void _intersectSpansRegion(const SwRleData *clip, SwSpan *spans, SwSpan *end, SwSpan **outSpans, uint32_t available)
+{
+    SwSpan *out = *outSpans;
+    int16_t sx1, sx2, cx1, cx2, x;
+    uint16_t len;
+    const SwSpan *clipSpans = clip->spans;
+    const SwSpan *clipEnd = clip->spans + clip->size;
+
+    while (available && spans < end ) {
+        if (clipSpans > clipEnd) {
+            spans = end;
+            break;
+        }
+        if (clipSpans->y > spans->y) {
+            ++spans;
+            continue;
+        }
+        if (spans->y != clipSpans->y) {
+            ++clipSpans;
+            continue;
+        }
+        sx1 = spans->x;
+        sx2 = sx1 + spans->len;
+        cx1 = clipSpans->x;
+        cx2 = cx1 + clipSpans->len;
+
+        if (cx1 < sx1 && cx2 < sx1) {
+            ++clipSpans;
+            continue;
+        }
+        else if (sx1 < cx1 && sx2 < cx1) {
+            ++spans;
+            continue;
+        }
+        x = sx1 > cx1 ? sx1 : cx1;
+        len = (sx2 < cx2 ? sx2 : cx2) - x;
+        if (len) {
+            uint16_t spansCorverage = spans->coverage;
+            uint16_t clipSpansCoverage = clipSpans->coverage;
+            out->x = sx1 > cx1 ? sx1 : cx1;
+            out->len = (sx2 < cx2 ? sx2 : cx2) - out->x;
+            out->y = spans->y;
+            out->coverage = (uint8_t)((spansCorverage * clipSpansCoverage) >> 8);
+            ++out;
+            --available;
+        }
+        if (sx2 < cx2) ++spans;
+        else ++clipSpans;
+    }
+    *outSpans = out;
+}
+
+void _intersectSpansRect(const SwBBox *bbox, const SwSpan *spans, const SwSpan *end, SwSpan **outSpans, uint32_t available)
+{
+    SwSpan *out = *outSpans;
+    int16_t minx, miny, maxx, maxy;
+    minx = bbox->min.x;
```

Comment:
auto minx = static_cast<int16_t>(bbox->min.x);
auto miny = ...
auto maxx = ...
auto maxy = ...

Reply:
changed it